src/get-started/first-java-service.md

```diff
@@ -0,0 +1,605 @@
+    The controller may encode it in any suitable format (e.g., JSON).
+
+Finally, connect the controller to the service. `MyService` already has
+an empty `createPublicApiHandlers` method, modify it to have:
+
+<!--codeinclude-->
+[MyService.createPublicApiHandlers](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-createPublicApiHandlers
+<!--/codeinclude-->
+
+!!! success
+    That's it with the service implementation! Package
+    the [_service artifact_](../glossary.md#artifact)
+    and run the integration tests:
+
+    ```shell
+    mvn verify
+    ```
+
+    and then proceed to the next section, where we will test its operation.
+
+## Test Network
+
+Let's now launch a test network in which we can see our service operation.
+Our project already has a script launching a test network with a single
+validator node: `start-testnet.sh`.
+
+Run the script:
+
+```shell
+chmod 744 start-testnet.sh # Allow the script execution, needed once
+./start-testnet.sh
+```
+
+When you see messages like the following, the network is active:
+
+<!-- markdownlint-disable line-length -->
+
+```text
+[2020-03-05T10:36:24Z INFO  exonum_node::consensus] COMMIT ====== height=4, proposer=0, round=1, committed=0, pool=0, hash=43ac20f8b...
+```
+
+<!-- markdownlint-enable line-length -->
+
+Open a separate shell session and check the active services:
+
+```shell
+# You may pipe the response into `jq` to pretty-print it, if you have it
+# installed:
+curl -s http://127.0.0.1:3000/api/services/supervisor/services # | jq
+```
+
+You can see in the output the lists of deployed _service artifacts_ and
+_service instances_.
+However, the network has neither our _service artifact_ nor
+an _instance_ of our service. That is natural, because the service
+must be registered in the network first, and then it may be instantiated.
+
+### Service Instantiation
+
+#### Install the Java Launcher
+
+To register a service artifact that we built previously in the network,
+we will need `exonum-launcher` tool. It is a Python application which we
+recommend to install in a [virtual environment][python-venv]:
+
+```shell
+python3 -m venv .venv
+source .venv/bin/activate
+```
+
+[python-venv]: https://docs.python.org/3/library/venv.html
+
+Then install the `exonum-launcher` with the Java runtime support:
+
+```shell
+pip install exonum-launcher-java-plugins
+```
+
+Check it works:
+
+```shell
+python -m exonum_launcher --help
+```
+
+#### Start a Test Instance
+
+Next, we shall place the service artifact into an artifacts directory
+of the node: `testnet/artifacts`.
+
+```shell
+# Create the artifacts directory
+mkdir testnet/artifacts
+# Copy the artifact
+cp car-registry-service/target/car-registry-service-1.0.0-artifact.jar \
+   testnet/artifacts/
+```
+
+<!--
+TODO: Shall the _node_ (= Java runtime) create an artifacts directory if one
```

Comment:
FYI, the `exonum-cli` crate does create missing directories (e.g., the `artifacts-dir` for the `run-dev` command) automatically if they are missing. However, there is a difference in that these dirs aren't supposed to be shared among multiple nodes.

Reply:
I see, thanks, the mentioned issue is only applicable if one launches several nodes locally, and specifies in their configuration the same artifacts directory. I guess the target file systems handle parallel requests for a new directory perfectly, so it might not be a concern.